presto-main/src/main/java/com/facebook/presto/cost/StatisticRange.java

```diff
@@ -36,8 +38,8 @@ public StatisticRange(double low, double high, double distinctValues)
     {
         checkState(low <= high || (isNaN(low) && isNaN(high)), "Low must be smaller or equal to high or range must be empty (NaN, NaN)");
         checkState(distinctValues >= 0 || isNaN(distinctValues), "Distinct values count cannot be negative");
-        this.low = low;
-        this.high = high;
+        this.low = isNaN(low) ? NEGATIVE_INFINITY : low;
```

Comment:
`NaN`are fine here. We only allow both `low` and `high` to be `NaN`. And that is representation of an empty range.
The problem here is that we do not have logic to transform `VARCHAR` domain to `double` domain used by comparisons in filters.

This PR does not seem like a proper solution. I will look into that.